src/SAWScript/Crucible/MIR/Builtins.hs

```diff
@@ -282,24 +282,36 @@ constructExpandedSetupValue cc sc = go
                     , show (PP.pretty ty)
                     ]
         StructShape ty _ fldShps ->
+          case ty of
+            Mir.TyAdt adtName _ _ ->
+              case col ^. Mir.adts . at adtName of
+                Just adt@(Mir.Adt _ kind _ _ _ _ _) ->
+                  case kind of
+                    Mir.Struct -> do
+                      flds <- goFlds pfx fldShps
+                      pure $ MS.SetupStruct adt flds
+                    Mir.Enum{} ->
+                      fail "`repr(transparent)` enums not currently supported"
+                    Mir.Union ->
+                      panic "constructExpandedSetupValue"
+                            [ "Unexpected `repr(transparent)` union:"
+                            , show adtName
+                            ]
```

Comment:
I know nothing about MIR so I might be wrong here, but are you sure these cases go under `StructShape` and not `TransparentShape`?

Reply:
Oops, good catch. Really, both cases only make sense for `Struct`, although I wrote the non-`Struct` errors in a bit of a hurry and forgot to put them in the right place. I've now moved them to their proper homes under `TransparentShape`, and I've added more precise errors under the `StructShape` case.